### PR TITLE
client: Add GetMetrics

### PR DESCRIPTION
**Pull request by stgraber**
Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>

### CODE REVIEW

**Comment by stgraber** [comment]
jenkins: test this please

File: client/lxd_server.go [hunk 1]
```diff
@@ -141,3 +143,41 @@ func (r *ProtocolLXD) UseTarget(name string) InstanceServer {
 func (r *ProtocolLXD) IsAgent() bool {
 	return r.server != nil && r.server.Environment.Server == "lxd-agent"
 }
+
+// GetMetrics returns the text OpenMetrics data.
+func (r *ProtocolLXD) GetMetrics() (string, error) {
```
Comment (tomponline) [inline client/lxd_server.go:148]: Can this be a more structured type? Like map[string]any ?

Comment (stgraber) [reply client/lxd_server.go:148]: No, it's not a map, it's a text document with headers and a pretty complex structure.
We could in theory send it through the prometheus parser and return internal go-prometheus structs instead but that feels pretty wrong.

In practice, it's an OpenMetrics Text document so returning it as a string seems appropriate.

Comment (tomponline) [reply client/lxd_server.go:148]: Ok fair enough.

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
jenkins: test this please

**Pull request merged**